utils/gen_string.h

```diff
@@ -0,0 +1,94 @@
+/*
+ * random string stream generator:
+ * generate a stream of string randomly with the length of the string between [len_min, len_max]
+ * Example: generating a string stream with 10 strings.
+ *   gen_string gen;
+ *   for (int i=0; i<10; i++) gen.next();
+ */
+
+#ifndef GEN_STRING_
+#define GEN_STRING_ 
+
+#include <iostream>
+#include <string>
+#include <random>
+
+class gen_string {
+
+  public:
+    /*
+     * default constructor 
+     * @seed=1 
+     * @min_len=6 
+     * @max_len=20
+     * @pool={'0',...,'9','A',...,'Z','a',...,'z'}
+     */
+    gen_string():seed(1),min_len(6),max_len(20) {
+      gen.seed(seed);
+      generate_pool();
+      dist_code=std::uniform_int_distribution<int>(0,pool.size()-1);
```

Comment:
Spaces around the `=`.

Did you run the clang formatter on this file? It still seems to have a few issues, like the spacing around the `=`, lines being longer than 80 columns etc.

Reply:
I forgot to format this file, sorry :sweat_smile:. Other files are formatted.